connect/src/authProviderRelay.ts

```diff
@@ -0,0 +1,66 @@
+
+import type { AuthProvider, LinkProof } from '@ceramicnetwork/blockchain-utils-linking'
+import { RPCClient } from 'rpc-utils'
+import { serveCrossOrigin } from '@ceramicnetwork/rpc-postmessage'
+import {  createClient } from '@ceramicnetwork/rpc-transport'
+import { PostMessageTransport, PostMessageTarget } from '@ceramicnetwork/transport-postmessage'
+import { AccountID } from 'caip'
+
+type AuthProviderMethods = {
+  accountId: { result: string },
+  authenticate: {
+    params: { message: string},
+    result: string
+  },
+  createLink: {
+    params: { did: string},
+    result: LinkProof
+  }
+}
+
+export class AuthProviderClient implements AuthProvider {
+
+  client: RPCClient
+  readonly isAuthProvider = true;
+
+  constructor(target:PostMessageTarget) {
+    target = target || window.parent
+    const transport = new PostMessageTransport<AuthProviderMethods>(window, target, {
+      postMessageArguments: [window.origin], 
+    })
+    //TODO Cant resolve these types???
+    this.client = createClient<AuthProviderMethods>(transport)
```

Comment:
@PaulLeCam  any idea why i cant resolve types properly here? ( transport)

Reply:
Yes the transport only cares about messages in/out so the methods are too high-level here, it needs something like:
```ts
type Request = RPCRequest<AuthProviderMethods, keyof AuthProviderMethods>
type Response = RPCResponse<AuthProviderMethods, keyof AuthProviderMethods>
const transport = new PostMessageTransport<Response, Request>(window, target, {
  postMessageArguments: [window.origin],
})
```

I'm working on updating this in a separate branch off this one, as we also need to use the latest version of rpc-utils to have the proper types, hopefully that should unblock you.